### PR TITLE
fix(views): prevent direct calls to an ajax view

### LINKED ISSUE #4959: Output of the 'ajax' page handler on non-xhr calls (Trac #4959)
_Original ticket http://trac.elgg.org/ticket/4959 on 42950404-06-10 by trac user ismayil.khayredinov, assigned to unknown._

Elgg version: 1.8

Calls to 'ajax' page handler with non XHR headers should either fail or be wrapped in a proper page handler.

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #4959

### CODE REVIEW

**Comment by mrclay** [comment]
@hypeJunction agreed. register_error, then forward.

File: languages/en.php [hunk 1]
```diff
@@ -41,6 +41,7 @@
 	'actionunauthorized' => 'You are unauthorized to perform this action',
 	
 	'ajax:error' => 'Unexpected error while performing an AJAX call. Maybe the connection to the server is lost.',
+	'ajax:not_is_xhr' => 'AJAX: It is not allowed to access these AJAX views directly.',
```
Comment (beck24) [inline languages/en.php]: The wording for this message seems clunky.

"You cannot access AJAX views directly"
"Accessing AJAX views directly is not allowed"
"AJAX views are not directly accessible"

?

**Comment by jdalsem** [comment]
if we use forward, we can not set the 400 bad request state

**Comment by hypeJunction** [comment]
Did anything change?

**Comment by hypeJunction** [comment]
Let's add more error handlers, 301 and 400 are most common.

**Comment by jdalsem** [comment]
> Did anything change?

Just the text for now....




> Let's add more error handlers, 301 and 400 are most common.

seperate ticket to solve that first?

**Comment by hypeJunction** [comment]
We don't need to wait. You can do the forward, the HTTP codes are in core, but there are no views that describe them.

**Comment by hypeJunction** [comment]
Eh. I am all confused with our error handling. The codes are there, but there are no hooks for anything other than 404.

**Comment by hypeJunction** [comment]
I will work on the error hooks

**Comment by jdalsem** [comment]
updated to use a forward...
@hypeJunction will you make a new ticket for the error handling?

**Comment by hypeJunction** [comment]
I think there is a ticket. Need to dig it out

**Pull request merged**